<a href="https://colab.research.google.com/github/ethankyle360/ML4SupplyChain/blob/main/Section_15_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Annual Demand for raw coffee: 6,000 tons

Price of 1 ton: 1,500 USD

Holding rate: 10%

Cost of transportation & order: 4,000 USD

---

What is the optimal Q?

What is the total logistics cost?

What is the t practical?

If the supplier will offer a 10% discount if your Q is 700, would I accept it?

If the lead time from the Columbian supplier is 2 months, what is the reorder point?


In [1]:
pip install inventorize1

  Preparing metadata (setup.py) ... done
  Created wheel for inventorize1: filename=inventorize1-0.0.2-py3-none-any.whl size=15612 sha256=c8d742f6bc3d39e40bfb61fe6e66783a52d4f2bcdd36f873db95586b8b5b8e56
  Stored in directory: /root/.cache/pip/wheels/88/ff/6d/cc065d3a5380732f2c1c06faae76d805a83ebc3c4412cd9eed
Successfully built inventorize1


In [2]:
import inventorize1 as inv
import pandas as pd

In [3]:
d = 6000 # Annual demand for raw coffee (in tons)
c = 1500 # Price of 1 ton
s = 4000 # Cost of transportation and order
h = 0.1 # Holding rate cost

Economic Order Quantity (EOQ) is a formula used to find the optimal quantity to order to minimize inventory costs.

In [4]:
eoq = pd.DataFrame(inv.eoq(d, s, c, h), index=[0])

`.loc` is used to access specific part from a DataFrame. This of loc as an acronym for locate/location.

Locate the value in the row index `0` and column label `EOQ`

In [5]:
# Get access to the DataFrame `eoq`
eoq1 = eoq.loc[0, 'EOQ']

Calculate the Total Logistics Cost (TLC) using the EOQ model.

`(d/eoq1) * s`: Calculates the total ordering cost

`(eoq1/2) * (h * c)`: Calculates the total holding cost

`c * d`: Calculates the total purchasing cost

In [6]:
TLC = (d/eoq1) * s + (eoq1/2) * (h*c) + c * d

In [7]:
inv.TQpractical(d, s, c, h)

{'Ts': 0.09428090415820634,
 'Tstarweeks': 4.90260701622673,
 'Qstar': 565.685424949238,
 'Tpractical': 4,
 'Tpracticalweeks': 4,
 'Qpractical': 461.53846153846155}

`Ts`: Place an order approximately every 0.094 years (every 34.4 days)

`Tstarweeks`: Place an order approximately every 4.9 weeks (every 34.4 days)

`Qstar`: Optimal number of lemonade supplies (lemons, sugar, cups) to order each time is 566 units to minimize costs.

`Tpractical`: It's practical to place an order every 4 months to align with supplier schedules.

`Tpracticalweeks`: It's best to place an order every 4 weeks (once a month).

`Qpractical`: For each ordering time (every 4 weeks), I'd order about 462 units of supplies each time.

Calculate the Total Logistics Cost (TLC) but with the following:
1. Fixed order quantity of 700 units
2. Reduced costs by 10%

In [12]:
### At 10% discount
## 0.9 represents a 10% reduction in both the holding and purchasing cost.
TLC1 = (d/700) * s + (700/2) * (h*c*0.9) + c * 0.9 * d
print("TLC1:", TLC1)

TLC1: 8181535.714285715


Conclusion: The total annual cost for ordering, holding, and purchasing the inventory, given a 10% reduction in holding and purchasing costs is approximately $8,181,535.71

In [15]:
## t represents cycle times in years
t = eoq1 / d
print("t:", t)
## L represents lead time in years
L = 2/12
print("L:", L)
## Check if lead time is greater than cycle time
L > t
## l_prime represents adjusted lead time
l_prime = L-(1*(eoq1/d))
print("l_prime:", l_prime)
## Calculate reorder points in units
reorder_point = l_prime * d
print("reorder_point:", reorder_point)

t: 0.09428090415820634
L: 0.16666666666666666
l_prime: 0.07238576250846032
reorder_point: 434.3145750507619


1. `t`: an order must be placed every 0.0942 years (which is every 34.4 days)
2. `L`: It takes about 0.167 years (2 months) from placing an order until the order inventory is received.
3. `l_prime`: After accounting 1 cycle time, the remaining lead time is 0.0724 years (about 26.4 days)
4. `reorder_point`: When the inventory level drops to 434.31 units, it's time to place a new order to ensure there is enough stock to cover the demand during the lead time.